packages/core/test/types.test.ts

```diff
@@ -229,16 +244,17 @@ describe('Raise events', () => {
 
 describe('types', () => {
   it('defined context in createMachine() should be an object', () => {
-    createMachine({
+    createMachine2({
       // @ts-expect-error
       context: 'string'
     });
   });
 });
 
 describe('context', () => {
-  it('should infer context type from `config.context` when there is no `schema.context`', () => {
-    createMachine(
+  // TODO: how would we do context inference here, if that is even desirable?
```

Comment:
Inference for a single slot is not possible with "generic maps". Only a type parameter can be inferred but we no longer have `TContext` here... we have `TTypes` and only that can be inferred (where `TTypes['context']` can't be inferred on its own)

Reply:
Agreed. That capability was built on top of TS tricks anyway - and I'm not super confident in them. It was here mainly to match closer how it used to work before `tsTypes` and typegen but it's better to just require this as part of the other "generics"